src/server/scripts/Spells/spell_evoker.cpp

```diff
@@ -73,7 +78,64 @@ class spell_evo_glide : public SpellScript
     }
 };
 
+// 361469 - Living Flame (Red)
+class spell_evo_living_flame : public SpellScript
+{
+    PrepareSpellScript(spell_evo_living_flame);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_EVOKER_LIVING_FLAME_DAMAGE, SPELL_EVOKER_LIVING_FLAME_HEAL });
+    }
+
+    void HandleTarget(SpellEffIndex /*effIndex*/)
+    {
+        if (GetCaster()->IsFriendlyTo(GetHitUnit()))
+            GetCaster()->CastSpell(GetHitUnit(), SPELL_EVOKER_LIVING_FLAME_HEAL, true);
+        else
+            GetCaster()->CastSpell(GetHitUnit(), SPELL_EVOKER_LIVING_FLAME_DAMAGE, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_evo_living_flame::HandleTarget, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 361500 - Living Flame (damage)
+class spell_evo_living_flame_damage : public SpellScript
+{
+    PrepareSpellScript(spell_evo_living_flame_damage);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_EVOKER_ENERGIZING_FLAME, SPELL_EVOKER_LIVING_FLAME });
+    }
+
+    void HandleManaRestored(SpellEffIndex /*effIndex*/)
+    {
+        if (AuraEffect* auraEffect = GetCaster()->GetAuraEffect(SPELL_EVOKER_ENERGIZING_FLAME, EFFECT_0))
+        {
+            SpellInfo const* spellInfo = sSpellMgr->AssertSpellInfo(SPELL_EVOKER_LIVING_FLAME, GetCastDifficulty());
+
+            Optional<SpellPowerCost> cost = spellInfo->CalcPowerCost(POWER_MANA, false, GetCaster(), GetSpellInfo()->GetSchoolMask(), nullptr);
+            if (!cost)
+                return;
+
+            int32 manaRestored = CalculatePct(cost->Amount, auraEffect->GetAmount());
```

Comment:
Do you restore mana based on the cost you did SPENT or the base cost of the spell?

Reply:
the spell costs 1000, so must restore 500 (50%)

Reply:
> the spell costs 1000, so must restore 500 (50%)

thats not what i meant.. what if you got a aura to reduce your mana cost by 50% and you only spend 500, should it restore 250 then or still 500?

Reply:
yeah 250, the 50% of the spell cost

Reply:
but your code is getting the cost of the spellInfo not of the spell that you casted

Reply:
the spell that costs mana is 361469, the damage spell only hits damage

Reply:
when casting with innervate up you do not regenerate any additional mana